### DISCUSSION TITLE
Sorting in Superset Heatmap Visualization

### DISCUSSION THREAD

**Discussion by lauslzrm** [apache/superset#30371] [Q&A / Help] [answered]
**Description:**

I'm using a heatmap visualization in Apache Superset, and I'm facing an issue with sorting the axes. Specifically, one of the axes represents the days of the week, but they are sorted alphabetically by default. I need them to be sorted chronologically from Monday to Sunday.

**What I’ve Tried:**

I attempted to use an additional column to specify the correct order of the days, but Superset does not allow me to use this column for sorting. The only option available in the settings is ‘sort axis’, which allows sorting alphabetically or by the selected metric. Neither of these options suits my needs, as I cannot use another metric to achieve the desired order.

As a workaround, I’ve been adding spaces to the day labels to force the correct order, using a CASE statement like this:
```
CASE numdayweek
  WHEN 1 THEN '      Sunday'
  WHEN 2 THEN 'Monday'
  WHEN 3 THEN ' Tuesday'
  WHEN 4 THEN '  Wednesday'
  WHEN 5 THEN '   Thursday'
  WHEN 6 THEN '    Friday'
  WHEN 7 THEN '     Saturday'
END
```
This approach technically works but feels like a hack and is not ideal for readability or maintainability.

**Request for Help:**

I’m looking for a cleaner, more efficient solution to sort the days of the week chronologically within the heatmap visualization. If anyone has faced a similar issue or knows of a better way to handle this in Superset, your insights would be greatly appreciated!

Thank you!

**Comment by sfirke** [comment, marked as answer]
I agree that this functionality is valuable and missing!  I have the same problem.  I've created a feature request here: https://github.com/apache/superset/discussions/30575  My case is about bar charts but I mentioned Heatmaps too because of this and linked back to this discussion.  See the workarounds section of my post, I linked to someone who is doing the same whitespace hack but maybe in a less-obvious, cleaner way.

Do you mind if I close this discussion just to concentrate discussion on a single issue?

**Reply by lauslzrm** [reply]
As far as I'm concerned, no problem at all @sfirke

**Comment by sfirke** [comment]
I can validate this problem applies to Heatmaps, here's one I just made - this x-axis is not good.
![image](https://github.com/user-attachments/assets/2ab56b39-5081-4d10-a87b-e288860d86bf)

**Comment by michofwood** [comment]
This should be done at the dataset level so that for a column, it is sorted by another one. That would apply to filters and projections and anywhere a column values are exposed.
All BI tools have that intrinsict feature built in at the Data model stage.
